simson/common/trade_balancers.py

```diff
@@ -0,0 +1,48 @@
+import sys
+
+
+def balance_by_extrenum(trade, by: str, inplace=False):
+    global_imports = trade.imports.sum_nda_over('r')
+    global_exports = trade.exports.sum_nda_over('r')
+
+    if by == 'maximum':
+        reference_trade = global_imports.maximum(global_exports)
+    elif by == 'minimum':
+        reference_trade = global_imports.minimum(global_exports)
+    elif by == 'imports':
+        reference_trade = global_imports
+    elif by == 'exports':
+        reference_trade = global_exports
+    else:
+        raise ValueError(f"Extrenum {by} not recognized. Must be one of "
+                         f"'maximum', 'minimum', 'imports' or 'exports'.")
+
```

Comment:
This could be a dict lookup
```suggestion
    reference_trade_lookup = {
        'maximum': global_imports.maximum(global_exports),
        'minimum': global_imports.minimum(global_exports),
        'imports': global_imports,
        'exports': global_exports,
    }
    if by not in reference_trade_lookup:
        raise ValueError(f"Extrenum {by} not recognized. Must be one of {list(reference_trade_lookup.keys())}")
    reference_trade = reference_trade_lookup[by]

```

Reply:
Good idea!